instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/engine/unit/UnitConverterFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jmx.engine.unit;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+public class UnitConverterFactory {
+
+  private static final Map<String, Map<String, UnitConverter>> conversionMappings = new HashMap<>();
+
+  static {
+    registerConverter("ms", "s", value -> value.doubleValue() / TimeUnit.SECONDS.toMillis(1), true);
+    registerConverter("ns", "s", value -> value.doubleValue() / TimeUnit.SECONDS.toNanos(1), true);
+  }
+
+  private UnitConverterFactory() {}
+
+  public static UnitConverter getConverter(@Nullable String fromUnit, @Nullable String toUnit) {
```

Comment:
`fromUnit` should be the one provided by `sourceUnit`, so I think only this one is likely to be `null`, the `toUnit` is mandatory and thus not annotated with `@Nullable`.